### PR TITLE
TST: Fix plotting tests

### LINKED ISSUE #1717: TST: properly test plotting
Moved from https://github.com/geopandas/geopandas/pull/1694#issuecomment-733044056

@gcaria's comment on some plotting tests:
 
It seems that tests for all geometric objects are already in place, but I've found something interesting.

The test  for `MultiPoint`s
https://github.com/geopandas/geopandas/blob/5856332704233fcd54696999d9ee4f16f1a1ebff/geopandas/tests/test_plotting.py#L295-L298

- only checks the first 2 of 20 plotted points;
- is missing the optional argument`lut=2` when calling `plt.get_cmap`.

The test for `MultiLineString`s seems alright at first glance
https://github.com/geopandas/geopandas/blob/59d3b2b7260b485e95e06a38f1c3e218b0a40a46/geopandas/tests/test_plotting.py#L525-L529

but I tried to run a standalone version of the test and found:

```python
N = 10
lines = GeoSeries([LineString([(0, i), (4, i + 0.5), (9, i)]) for i in range(N)],
                               index=list("ABCDEFGHIJ"),)

multiline1 = MultiLineString(lines.loc["A":"B"].values)
multiline2 = MultiLineString(lines.loc["C":"D"].values)
df = GeoDataFrame({"geometry": [multiline1, multiline2], "values": [0, 1]})

ax = df.plot(column="values")
ax.collections[0].get_facecolors()

# returns array([], shape=(0, 4), dtype=float64)
```
which means that the `zip` output used in `test_plotting._check_colors` is empty.

I've then run a quick test (...of the tests) printing the `len`s of the `zip` inputs within `test_plotting._check_colors` and found mismatches in these tests (sometimes the first input having `len=0` as in `test_multilinestrings`):

```
geopandas/tests/test_plotting.py::TestPointPlotting::test_colormap
geopandas/tests/test_plotting.py::TestLineStringPlotting::test_multilinestrings
geopandas/tests/test_plotting.py::TestPolygonPlotting::test_single_color
geopandas/tests/test_plotting.py::TestGeometryCollectionPlotting::test_colors
geopandas/tests/test_plotting.py::TestGeometryCollectionPlotting::test_values
geopandas/tests/test_plotting.py::TestPlotCollections::test_linestrings_values
geopandas/tests/test_plotting.py::TestPlotCollections::test_polygons
geopandas/tests/test_plotting.py::TestPlotCollections::test_polygons_values
```

### PR DESCRIPTION
**Pull request by gcaria**
Fixes #1717

I've fixed the tests where there were different lengths of actual and expected colors for the `_check_colors` function (sometimes a length was zero which means there was nothing actually being checked).
In order to avoid this situation I've also added a simple `assert` in `_check_colors`.

Please let me know if there is something that doesn't make sense even though the tests "are working".

### CODE REVIEW

File: geopandas/tests/test_plotting.py [hunk 1]
```diff
@@ -845,17 +845,27 @@ def setup_method(self):
...
-        _check_colors(2, ax.collections[1].get_edgecolors(), [MPL_DFT_COLOR])  # line
-        _check_colors(2, ax.collections[2].get_facecolors(), [MPL_DFT_COLOR])  # point
+        _check_colors(
+            2, ax.collections[0].get_facecolors(), [MPL_DFT_COLOR] * 2
+        )  # poly
+        _check_colors(
+            2, ax.collections[1].get_edgecolors(), [MPL_DFT_COLOR] * 2
+        )  # line
+        _check_colors(
+            2, ax.collections[2].get_facecolors(), [MPL_DFT_COLOR] * 2
```
Comment (martinfleis) [inline geopandas/tests/test_plotting.py:855]: There's only one point in the `self.series`, should not this be `1, ax.collections[2].get_facecolors(), [MPL_DFT_COLOR] `?

File: geopandas/tests/test_plotting.py [hunk 2]
```diff
@@ -845,17 +845,27 @@ def setup_method(self):
...
-        _check_colors(1, ax.collections[0].get_facecolors(), exp_colors)  # poly
-        _check_colors(2, ax.collections[1].get_edgecolors(), [exp_colors[0]])  # line
-        _check_colors(2, ax.collections[2].get_facecolors(), [exp_colors[1]])  # point
+        exp_colors = cmap([0.0, 1.0])
+        _check_colors(2, ax.collections[0].get_facecolors(), exp_colors)  # poly
+        _check_colors(
+            2, ax.collections[1].get_edgecolors(), [exp_colors[0]] * 2
+        )  # line
+        _check_colors(
+            2, ax.collections[2].get_facecolors(), [exp_colors[1]] * 2
```
Comment (martinfleis) [inline geopandas/tests/test_plotting.py:867]: Same as above but I am probably just a bit confused here.

**Review by martinfleis — COMMENTED** [review]
Thanks a lot for this! I have left just one question/comment in the code but it is essentially ready to be merged.

File: geopandas/tests/test_plotting.py (hunk 1 above)
Comment (gcaria) [reply geopandas/tests/test_plotting.py:855]: That's right, thanks for pointing that out !

**Review by jorisvandenbossche — APPROVED** [review]
Thanks a lot for cleaning this up!

**Comment by jorisvandenbossche** [comment]
Failures are the pandas extension tests that are fixed on master.

**Pull request merged**